### PR TITLE
update ui to shadcn and more

### PR DESCRIPTION
**Pull request by Joshimello**

### CODE REVIEW

File: tailwind.config.ts [hunk 1]
```diff
@@ -1,19 +1,80 @@
...
+        "accordion-up": "accordion-up 0.2s ease-out",
+      },
     },
   },
-  plugins: [
-    require('@tailwindcss/typography'),
-  ],
-}
-export default config
+  plugins: [require("tailwindcss-animate")],
```
Comment (ImJustChew) [inline tailwind.config.ts:77]:
```
require('@tailwindcss/typography'),
```

File: src/components/Header.tsx [hunk 2]
```diff
@@ -1,45 +1,41 @@
...
+        <p className="text-xs text-gray-600 dark:text-gray-400">
+        {language == 'en' &&
+          (currentSemester ?`AC${currentSemester.year} Sem ${currentSemester.semester}, Week ${currentWeek}`: `Holiday`)}
+        {language == 'zh' &&
+          (currentSemester ?`${currentSemester.year-1911}-${currentSemester.semester} 學期, 第${currentWeek}週`: `假期`)}
+        </p>
+      </div>
+      <div className="flex gap-2">
+        <Help/>
+        <Button variant="outline" asChild>
```
Comment (ImJustChew) [inline src/components/Header.tsx:30]: buttons seem a bit big?

File: src/app/[lang]/globals.css [hunk 3]
```diff
@@ -48,3 +48,76 @@ html, body {
...
+    --destructive-foreground: 210 40% 98%;
+
+    --border: 214.3 31.8% 91.4%;
+    --input: 214.3 31.8% 91.4%;
+    --ring: 222.2 84% 4.9%;
+ 
+    --radius: 0.5rem;
+  }
+ 
+  .dark {
```
Comment (ImJustChew) [inline src/app/[lang]/globals.css:85]: dark mode colors are bad, don't match up with existing theme

File: components.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  "tsx": true,
+  "tailwind": {
+    "config": "tailwind.config.ts",
+    "css": "src/app/[lang]/globals.css",
+    "baseColor": "slate",
+    "cssVariables": true,
+    "prefix": ""
+  },
+  "aliases": {
+    "components": "@/scomponents",
```
Comment (ImJustChew) [inline components.json:14]: why not in @/components tho?

**Review by ImJustChew — CHANGES_REQUESTED** [review]

File: src/app/[lang]/globals.css (hunk 3 above)
Comment (Joshimello) [reply src/app/[lang]/globals.css:76]: no shadcn default themes match up with mui theme,
picked a temp color that matches best for now.
ultimate fix is to remove all mui components for shadcn.

**Review by ImJustChew — APPROVED** [review]

**Pull request merged**